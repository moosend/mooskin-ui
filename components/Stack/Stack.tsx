import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { IStackComponentProps } from './model';

// Components
import { Box } from '../Box/Box';

// Styled Components
import { StyledStack } from './styles';

/**
 * Stack
 */
export const Stack: React.FC<IStackComponentProps> = withMooskinContext((props) => {
	const getDividerSpacing = () => {
		switch (props.direction) {
			case 'row':
			case 'row-reverse':
				return { mx: props.spacing, alignSelf: 'stretch' as 'stretch' };

			case 'column':
			case 'column-reverse':
				return { my: props.spacing };

			default:
				return {};
		}
	};

	const getChildSpacing = (index: number) => {
		switch (props.direction) {
			case 'row':
				return index ? { marginLeft: props.spacing } : {};
			case 'row-reverse':
				return index ? { marginRight: props.spacing } : {};

			case 'column':
				return index ? { marginTop: props.spacing } : {};
			case 'column-reverse':
				return index ? { marginBottom: props.spacing } : {};

			default:
				return {};
		}
	};

	const dividerSpacingHandler = (element: JSX.Element | React.ReactElement, i: number) => {
		const spacingStyles = props.spacing ? getDividerSpacing() : {};
		return <Box {...spacingStyles}>{React.cloneElement(element, { key: `divider${i}` })}</Box>;
	};

	const childSpacingHandler = (element: JSX.Element | React.ReactElement, i: number) => {
		const margin = props.spacing ? getChildSpacing(i) : {};
		return React.cloneElement(element, { key: `divider${i}`, style: { ...element.props.style, ...margin } });
	};

	const renderChildren = () => {
		return React.Children.map(props.children, (child, key) => {
			if (React.isValidElement(child)) {
				if (props.divider) {
					return (
						<>
							{key !== 0 && dividerSpacingHandler(props.divider, key)}
							{React.cloneElement(child, { key })}
						</>
					);
				}
				return childSpacingHandler(child, key);
			}
			return <div />;
		});
	};

	return <StyledStack {...props} children={renderChildren()} />;
});

Stack.displayName = 'Stack';

/**
 * HStack
 */
export const HStack: React.FC<IStackComponentProps> = withMooskinContext((props) => {
	return <Stack {...props} />;
});

HStack.defaultProps = {
	align: 'stretch',
	className: '',
	direction: 'row',
	flexWrap: 'wrap',
	justify: 'initial',
	spacing: 0,
	style: {},
};

HStack.displayName = 'HStack';

/**
 * VStack
 */
export const VStack: React.FC<IStackComponentProps> = withMooskinContext((props) => {
	return <Stack {...props} />;
});

VStack.defaultProps = {
	align: 'initial',
	className: '',
	direction: 'column',
	flexWrap: 'wrap',
	justify: 'center',
	spacing: 0,
	style: {},
};

VStack.displayName = 'VStack';
