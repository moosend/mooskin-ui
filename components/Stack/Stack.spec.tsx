import * as React from 'react';
import { HStack, VStack } from './Stack';

import { mount } from 'enzyme';

const boxStyle = {
	height: 40,
	width: 40,
};

describe('Stack', () => {
	test('renders Stack correctly', () => {
		const tree = mount(
			<VStack spacing={20} divider={<span style={{ width: 1 }} />}>
				<div key={0} style={{ ...boxStyle, backgroundColor: 'red' }} />,
				<div key={1} style={{ ...boxStyle, backgroundColor: 'green' }} />,
				<div key={2} style={{ ...boxStyle, backgroundColor: 'blue' }} />
			</VStack>
		);

		expect(tree).toMatchSnapshot();
	});

	test('renders Stack with spacings correctly', () => {
		const tree = mount(
			<VStack spacing={20}>
				<span key={0} style={{ ...boxStyle, backgroundColor: 'red' }} />,
				<span key={1} style={{ ...boxStyle, backgroundColor: 'green' }} />,
				<span key={2} style={{ ...boxStyle, backgroundColor: 'blue' }} />
			</VStack>
		);

		expect(tree.find('span').at(0).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'red' });
		expect(tree.find('span').at(1).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'green', marginTop: 20 });
		expect(tree.find('span').at(2).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'blue', marginTop: 20 });

		tree.setProps({ direction: 'column-reverse' });

		expect(tree.find('span').at(0).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'red' });
		expect(tree.find('span').at(1).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'green', marginBottom: 20 });
		expect(tree.find('span').at(2).prop('style')).toEqual({ ...boxStyle, backgroundColor: 'blue', marginBottom: 20 });
	});

	test('renders Stack with divider & spacings correctly', () => {
		const tree = mount(
			<HStack spacing={20} divider={<span style={{ width: 1 }} />}>
				<label key={0} style={{ ...boxStyle, backgroundColor: 'red' }} />,
				<label key={1} style={{ ...boxStyle, backgroundColor: 'green' }} />,
				<label key={2} style={{ ...boxStyle, backgroundColor: 'blue' }} />
			</HStack>
		);

		expect(tree.find('span').length).toEqual(2);
		// expect(tree.find('Box').first().prop('style')).toEqual({margin: '0 20px', alignSelf: 'stretch'});
	});
});
