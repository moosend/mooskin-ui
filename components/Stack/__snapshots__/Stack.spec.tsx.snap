// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack renders Stack correctly 1`] = `
<VStack
  align="initial"
  className=""
  direction="column"
  divider={
    <span
      style={
        Object {
          "width": 1,
        }
      }
    />
  }
  flexWrap="wrap"
  justify="center"
  spacing={20}
  style={Object {}}
>
  <Component
    align="initial"
    className=""
    direction="column"
    divider={
      <span
        style={
          Object {
            "width": 1,
          }
        }
      />
    }
    flexWrap="wrap"
    justify="center"
    palette={
      Object {
        "backgroundColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    spacing={20}
    style={Object {}}
  >
    <Stack
      align="initial"
      className=""
      direction="column"
      divider={
        <span
          style={
            Object {
              "width": 1,
            }
          }
        />
      }
      flexWrap="wrap"
      justify="center"
      palette={
        Object {
          "backgroundColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      spacing={20}
      style={Object {}}
    >
      <Component
        align="initial"
        className=""
        direction="column"
        divider={
          <span
            style={
              Object {
                "width": 1,
              }
            }
          />
        }
        flexWrap="wrap"
        justify="center"
        palette={
          Object {
            "backgroundColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        spacing={20}
        style={Object {}}
      >
        <StyledStack
          align="initial"
          className=""
          direction="column"
          divider={
            <span
              style={
                Object {
                  "width": 1,
                }
              }
            />
          }
          flexWrap="wrap"
          justify="center"
          palette={
            Object {
              "backgroundColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          spacing={20}
          style={Object {}}
        >
          <Box
            align="initial"
            className="sc-gtssRu iXkelt"
            direction="column"
            divider={
              <span
                style={
                  Object {
                    "width": 1,
                  }
                }
              />
            }
            flexWrap="wrap"
            justify="center"
            palette={
              Object {
                "backgroundColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            spacing={20}
            style={Object {}}
            transition="0.2s all"
          >
            <Component
              align="initial"
              className="sc-gtssRu iXkelt"
              direction="column"
              divider={
                <span
                  style={
                    Object {
                      "width": 1,
                    }
                  }
                />
              }
              flexWrap="wrap"
              justify="center"
              palette={
                Object {
                  "backgroundColors": Object {
                    "background": "#FAFAFA",
                    "black": "#000000",
                    "darkgray1": "#777777",
                    "darkgray2": "#454545",
                    "gray1": "#E0E0E0",
                    "gray2": "#E7E7E7",
                    "medgray1": "#757575",
                    "medgray2": "#888888",
                    "primary1": "#4169E1",
                    "primary2": "#212121",
                    "primary3": "#00796B",
                    "primary4": "#D32F2F",
                    "secondary1": "#2E7D32",
                    "secondary2": "#757575",
                    "secondary3": "#673AB7",
                    "secondary4": "#9C27B0",
                    "secondary5": "#00796B",
                    "secondary6": "#ECF0FC",
                    "text": "#212121",
                    "white": "#FFFFFF",
                  },
                  "borderColors": Object {
                    "background": "#FAFAFA",
                    "black": "#000000",
                    "darkgray1": "#777777",
                    "darkgray2": "#454545",
                    "gray1": "#E0E0E0",
                    "gray2": "#E7E7E7",
                    "medgray1": "#757575",
                    "medgray2": "#888888",
                    "primary1": "#4169E1",
                    "primary2": "#212121",
                    "primary3": "#00796B",
                    "primary4": "#D32F2F",
                    "secondary1": "#2E7D32",
                    "secondary2": "#757575",
                    "secondary3": "#673AB7",
                    "secondary4": "#9C27B0",
                    "secondary5": "#00796B",
                    "secondary6": "#ECF0FC",
                    "text": "#212121",
                    "white": "#FFFFFF",
                  },
                  "fontColors": Object {
                    "background": "#FAFAFA",
                    "black": "#000000",
                    "darkgray1": "#777777",
                    "darkgray2": "#454545",
                    "gray1": "#E0E0E0",
                    "gray2": "#E7E7E7",
                    "medgray1": "#757575",
                    "medgray2": "#888888",
                    "primary1": "#4169E1",
                    "primary2": "#212121",
                    "primary3": "#00796B",
                    "primary4": "#D32F2F",
                    "secondary1": "#2E7D32",
                    "secondary2": "#757575",
                    "secondary3": "#673AB7",
                    "secondary4": "#9C27B0",
                    "secondary5": "#00796B",
                    "secondary6": "#ECF0FC",
                    "text": "#212121",
                    "white": "#FFFFFF",
                  },
                }
              }
              spacing={20}
              style={Object {}}
              transition="0.2s all"
            >
              <StyledBox
                align="initial"
                className="sc-gtssRu iXkelt"
                direction="column"
                divider={
                  <span
                    style={
                      Object {
                        "width": 1,
                      }
                    }
                  />
                }
                flexWrap="wrap"
                justify="center"
                palette={
                  Object {
                    "backgroundColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                    "borderColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                    "fontColors": Object {
                      "background": "#FAFAFA",
                      "black": "#000000",
                      "darkgray1": "#777777",
                      "darkgray2": "#454545",
                      "gray1": "#E0E0E0",
                      "gray2": "#E7E7E7",
                      "medgray1": "#757575",
                      "medgray2": "#888888",
                      "primary1": "#4169E1",
                      "primary2": "#212121",
                      "primary3": "#00796B",
                      "primary4": "#D32F2F",
                      "secondary1": "#2E7D32",
                      "secondary2": "#757575",
                      "secondary3": "#673AB7",
                      "secondary4": "#9C27B0",
                      "secondary5": "#00796B",
                      "secondary6": "#ECF0FC",
                      "text": "#212121",
                      "white": "#FFFFFF",
                    },
                  }
                }
                spacing={20}
                style={Object {}}
                transition="0.2s all"
              >
                <div
                  className="sc-bdnylx cshEil sc-gtssRu iXkelt"
                  direction="column"
                  spacing={20}
                  style={Object {}}
                >
                  <div
                    key="0"
                    style={
                      Object {
                        "backgroundColor": "red",
                        "height": 40,
                        "width": 40,
                      }
                    }
                  />
                  <div
                    key=".1"
                  />
                  <Box
                    className=""
                    my={20}
                    style={Object {}}
                    transition="0.2s all"
                  >
                    <Component
                      className=""
                      my={20}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "borderColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "fontColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                        }
                      }
                      style={Object {}}
                      transition="0.2s all"
                    >
                      <StyledBox
                        className=""
                        my={20}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "borderColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "fontColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                          }
                        }
                        style={Object {}}
                        transition="0.2s all"
                      >
                        <div
                          className="sc-bdnylx ggKiRj"
                          style={Object {}}
                        >
                          <span
                            key="divider2"
                            style={
                              Object {
                                "width": 1,
                              }
                            }
                          />
                        </div>
                      </StyledBox>
                    </Component>
                  </Box>
                  <div
                    key="2"
                    style={
                      Object {
                        "backgroundColor": "green",
                        "height": 40,
                        "width": 40,
                      }
                    }
                  />
                  <div
                    key=".3"
                  />
                  <Box
                    className=""
                    my={20}
                    style={Object {}}
                    transition="0.2s all"
                  >
                    <Component
                      className=""
                      my={20}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "borderColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                          "fontColors": Object {
                            "background": "#FAFAFA",
                            "black": "#000000",
                            "darkgray1": "#777777",
                            "darkgray2": "#454545",
                            "gray1": "#E0E0E0",
                            "gray2": "#E7E7E7",
                            "medgray1": "#757575",
                            "medgray2": "#888888",
                            "primary1": "#4169E1",
                            "primary2": "#212121",
                            "primary3": "#00796B",
                            "primary4": "#D32F2F",
                            "secondary1": "#2E7D32",
                            "secondary2": "#757575",
                            "secondary3": "#673AB7",
                            "secondary4": "#9C27B0",
                            "secondary5": "#00796B",
                            "secondary6": "#ECF0FC",
                            "text": "#212121",
                            "white": "#FFFFFF",
                          },
                        }
                      }
                      style={Object {}}
                      transition="0.2s all"
                    >
                      <StyledBox
                        className=""
                        my={20}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "borderColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                            "fontColors": Object {
                              "background": "#FAFAFA",
                              "black": "#000000",
                              "darkgray1": "#777777",
                              "darkgray2": "#454545",
                              "gray1": "#E0E0E0",
                              "gray2": "#E7E7E7",
                              "medgray1": "#757575",
                              "medgray2": "#888888",
                              "primary1": "#4169E1",
                              "primary2": "#212121",
                              "primary3": "#00796B",
                              "primary4": "#D32F2F",
                              "secondary1": "#2E7D32",
                              "secondary2": "#757575",
                              "secondary3": "#673AB7",
                              "secondary4": "#9C27B0",
                              "secondary5": "#00796B",
                              "secondary6": "#ECF0FC",
                              "text": "#212121",
                              "white": "#FFFFFF",
                            },
                          }
                        }
                        style={Object {}}
                        transition="0.2s all"
                      >
                        <div
                          className="sc-bdnylx ggKiRj"
                          style={Object {}}
                        >
                          <span
                            key="divider4"
                            style={
                              Object {
                                "width": 1,
                              }
                            }
                          />
                        </div>
                      </StyledBox>
                    </Component>
                  </Box>
                  <div
                    key="4"
                    style={
                      Object {
                        "backgroundColor": "blue",
                        "height": 40,
                        "width": 40,
                      }
                    }
                  />
                </div>
              </StyledBox>
            </Component>
          </Box>
        </StyledStack>
      </Component>
    </Stack>
  </Component>
</VStack>
`;
